addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/NetatmoHandlerFactory.java

```diff
@@ -94,28 +101,56 @@ protected ThingHandler createHandler(Thing thing) {
         }
     }
 
+    @Override
+    protected void removeHandler(ThingHandler thingHandler) {
+        if (thingHandler instanceof NetatmoBridgeHandler) {
+            unregisterDeviceDiscoveryService(thingHandler.getThing().getUID());
+            unregisterWebHookServlet(thingHandler.getThing().getUID());
+        }
+        super.removeHandler(thingHandler);
+    }
+
     private void registerDeviceDiscoveryService(@NonNull NetatmoBridgeHandler netatmoBridgeHandler) {
-        NetatmoModuleDiscoveryService discoveryService = new NetatmoModuleDiscoveryService(netatmoBridgeHandler);
         if (bundleContext != null) {
-            discoveryServiceReg = bundleContext.registerService(DiscoveryService.class.getName(), discoveryService,
-                    new Hashtable<>());
+            NetatmoModuleDiscoveryService discoveryService = new NetatmoModuleDiscoveryService(netatmoBridgeHandler);
+            discoveryServiceRegs.put(netatmoBridgeHandler.getThing().getUID(), bundleContext.registerService(
+                    DiscoveryService.class.getName(), discoveryService, new Hashtable<String, Object>()));
         }
     }
 
-    @Override
-    protected void removeHandler(ThingHandler thingHandler) {
-        if (discoveryServiceReg != null && APIBRIDGE_THING_TYPE.equals(thingHandler.getThing().getThingTypeUID())) {
-            discoveryServiceReg.unregister();
-            discoveryServiceReg = null;
+    private void unregisterDeviceDiscoveryService(ThingUID thingUID) {
+        ServiceRegistration<?> serviceReg = discoveryServiceRegs.get(thingUID);
+        if (serviceReg != null) {
+            serviceReg.unregister();
+            discoveryServiceRegs.remove(thingUID);
         }
-        super.removeHandler(thingHandler);
     }
 
-    public void webHookEvent(NAWebhookCameraEvent event) {
-        logger.debug("Event transmitted from restService");
-        if (bridgeHandler != null) {
-            bridgeHandler.webHookEvent(event);
+    private WelcomeWebHookServlet registerWebHookServlet(ThingUID thingUID) {
+        WelcomeWebHookServlet servlet = null;
+        if (bundleContext != null) {
+            servlet = new WelcomeWebHookServlet(this.httpService, thingUID.getId());
```

Comment:
Please remove the prepended `this`. It doesn't fit into the code style of this binding.

Reply:
Done